akka-javasdk/src/main/java/akka/javasdk/agent/Agent.java

```diff
@@ -122,6 +123,34 @@ interface Builder {
        */
       Builder tools(List<Object> toolInstances);
 
+      /**
+       * Adds one or more tool classes that the AI model can use.
+       * <p>
+       * Each class must have at least one public method annotated with {@code @FunctionTool}.
+       * These methods will be available as tools for the AI model to invoke.
+       * <p>
+       * The object will be instantiated using the {@link DependencyProvider}.
+       * If no provider is defined, the class must have a no-argument default constructor.
+       *
+       * @param toolClasses additional tool classes exposing tool methods
+       * @return this builder for method chaining
+       */
+      Builder toolClasses(Class<?>... toolClasses);
+
+      /**
+       * Adds one or more tool classes that the AI model can use.
+       * <p>
+       * Each class must have at least one public method annotated with {@code @FunctionTool}.
+       * These methods will be available as tools for the AI model to invoke.
+       * <p>
+       * The object will be instantiated using the {@link DependencyProvider}.
+       * If no provider is defined, the class must have a no-argument default constructor.
+       *
+       * @param toolClasses one or more classes exposing tool methods
+       * @return this builder for method chaining
+       */
+      Builder toolClasses(List<Class<?>> toolClasses);
```

Comment:
Do we need the separation between tool classes and instances in the API? It could all be `Object`, and then at runtime we can see if it's a class or an instance.

Reply:
I didn't think about it. I will give a try.